22.1 reports (#170)
DevExpress/dotnet-eud: interface-elements-for-web/articles/report-designer/bind-to-data/bind-a-report-to-a-union-based-federated-data-source.md

---
title: Bind a Report to a Union-Based Federated Data Source
author: Boris Zaitsev
---

> # Bind a Report to Union-Based Federated Data Source

Reviewer: ```suggestion
# Bind a Report to a Union-Based Federated Data Source
```

    ![](../../../images/eurd-web-federated-datasource-union-json-datasource.png)

## Create Data Federation

1. Invoke the designer [menu](../report-designer-tools/menu.md) and click **Add Data Source**.

   ![](../../../images/eurd-web-data-source-menu.png) 

1. Select **Data Federation** in the invoked [Data Source Wizard](../report-designer-tools/data-source-wizard.md) and click **Next**.

    ![](../../../images/eurd-web-data-source-wizard-select-data-federation.png)

1. On the next page, click the **+ (plus) button** and select **Union** to add a Union query.

    ![](../../../images/eurd-web-data-federation-wizard-add-union-query.png)

1. In the invoked [Query Builder](../../../articles/report-designer/report-designer-tools/query-builder.md), drag and drop the table from the SQL data source onto the design surface.

    ![](../../../images/eurd-web-data-federation-union-query-builder-drop-table.png)

1. Drag and drop the JSON data source onto the design surface.

    ![](../../../images/eurd-web-data-federation-union-query-builder-drop-json-table.png)

> 1. The query includes only fields that have identical names and types in the original sources. You can specify a different name for a field. Click the field's **Alias** cell and enter the  new field name, if necessary. 

Reviewer: ```suggestion
1. The query includes only fields that have identical names and types in the original sources. You can specify a different name for a field. Click the field's **Alias** cell and enter the new field name, if necessary.
```

>     You can select the **Union All** checkbox in the bottom left to not remove duplicated entries from the query result.

Reviewer: Do you mean the following?

`You can select the **Union All** checkbox in the bottom left to prevent the removal of duplicate entries from the query result.`

Author: ```suggestion
    You can select the **Union All** checkbox in the bottom left to prevent the removal of duplicate entries from the query result.
```
